api/infrastructure/repository/order.repository.go

```diff
@@ -0,0 +1,32 @@
+package repository
+
+import (
+	"context"
+	"errors"
+
+	"gorm.io/gorm"
+	"shanepee.com/api/domain"
+)
+
+type orderRepositoryImpl struct {
+	db *gorm.DB
+}
+
+func (r *orderRepositoryImpl) FindOrdersByStatus(ctx context.Context, status string, sellerID int64) ([]*domain.Order, error) {
+	var order []*domain.Order
+	if err := r.db.Where("seller_id = ? AND status = ?", sellerID, status).Find(&order).Error; err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
+			return nil, domain.ErrOrderNotFound
+		}
+		return nil, err
+	}
+	return order, nil
```

Comment:
return empty array, dont return found

Reply:
```suggestion
	if err := r.db.Where("seller_id = ? AND status = ?", sellerID, status).Find(&order).Error; err != nil {
		return nil, err
	}
	return order, nil
```

i forget find cant return ErrRecordNotFound, so just `return nil, err` is fine kub